### DISCUSSION TITLE
LDAP troubleshooting ?

### DISCUSSION THREAD

**Discussion by OlivierBondu** [firefly-iii/firefly-iii#5074] [Firefly III] [answered]
Hello there !

I have trouble with my LDAP setup, I end up with either with a 'can't contact ldap server (-1)' or a 'Inappropriate authentication'.
I suspect the env variables not to be properly setup because running netcat within the container it looks the tcp ports 389 and 636 of the LDAP server are correctly opened...

I found around that executing within docker the `php artisan adldap:import -v` can help, however it does not look like the command read the latest environment variables whenever I change them.

--> Would anyone know a better way of troubleshooting my LDAP setup without restarting the container every single time ?

Thanks !

**Comment by OlivierBondu** [comment, marked as answer]
For the record, I managed to sort out my own ldap mess using `ldapsearch` - and in particular [this link](https://devconnected.com/how-to-search-ldap-using-ldapsearch-examples/)

**Reply by JC5** [reply]
Thanks for getting back! I must admit I forgot about your ticket entirely. :(

**Reply by OlivierBondu** [reply]
haha, no  worries, I did find out that my problem was ldap config, but bumped into an issue that [looks like this one](https://github.com/firefly-iii/firefly-iii/issues/5109) in the end... so I'll try right away the fresh 5.6.1 !

**Reply by JC5** [reply]
Cool! Curious to hear if it works as expected!

**Reply by OlivierBondu** [reply]
It does!
It still looks like I have trouble with LDAPS, but it works without TLS so that's a problem I'll postpone for now :)

thanks !

**Reply by JC5** [reply]
Let me know, I believe the package should support it.

**Reply by OlivierBondu** [reply]
not sure, but maybe my troubles are related to the expiration of Let's Encrypt ROOT CA ? Not sure how to check this...

https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/
